imap/mboxlist.c

```diff
@@ -1849,11 +1861,26 @@ EXPORTED int mboxlist_createmailbox_opts(const char *name, int mbtype,
         }
     }
 
-    r = mboxlist_createmailbox_full(name, mbtype, partition,
-                                    isadmin, userid, auth_state,
-                                    options, uidvalidity, createdmodseq, 0, 0, NULL,
-                                    uniqueid, localonly,
-                                    forceuser, dbonly, 0, &mailbox);
+    const mbentry_t mbentry = {
```

Comment:
Oh god.  Really?  I'd be tempted to go wtih mbentry = MBENTRY_INITIALIZER; then explicitly set the fields.  It's fewer lines and more "mbentry gets changed later" safe.